numpy/lib/stride_tricks.py

```diff
@@ -197,6 +197,37 @@ def _broadcast_shape(*args):
     return b.shape
 
 
+def _broadcast_shape_dispatcher(*args):
+    return args
+
+
+@array_function_dispatch(_broadcast_shape_dispatcher, module='numpy')
```

Comment:
I think there would be no need to possibly dispatch on any of the entries - as shapes in the form of arrays are not allowed anyway.

Otherwise, very nice to get this (we have our own python implementation in astropy that I'd happily get rid of).

p.s. The trick to use `np.empty(..., dtype=[])` could also be used inside `_broadcast_shape`. Maybe worth doing while you're at it?

Reply:
Agreed, dispatching is pointless here.

Reply:
Got it thank you - just pushed a commit to get rid of the array_function_dispatch.
 
@mhvk, _broadcast_shape operates on arrays, would the idea be to extract the shapes from the array arguments and use np.empty(..., dtype=[]) to call np.broadcast with empty arrays instead? I'm happy to make related improvements, but not sure I understand.

Reply:
It is just that I had not seen the `np.empty(shape, dtype=[])` trick before and inside `_broadcast_shape`, one has: https://github.com/numpy/numpy/blob/588a517b68d9f273b607622dc854b1b3d014ac58/numpy/lib/stride_tricks.py#L192-L195

It seemed one could use `b = np.empty(b.shape, dtype=[])` there as well (just checked; it is about a factor 10 faster).

Reply:
Ah got it, cool. Yes, I'm very happy to make this change, but perhaps should open a separate PR for it? (Thoughts, @eric-wieser?)